src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/toolWindow/UnrealToolWindowFactory.kt

```diff
@@ -232,95 +201,58 @@ class UnrealToolWindowFactory(val project: Project)
-                            if (UnrealPane.timestampCheckBox.isSelected) {
-                                startOfLine += TIME_WIDTH + 1
-                                endOfLine += TIME_WIDTH + 1
-                            }
-                            // we need to flush here if range doesn't exist
-                            if (endOfLine > consoleView.text.length) {
-                                consoleView.flushDeferredText()
-                            }
-                            run {
-                                if (currentRevision != UnrealPane.revision) {
-                                    return@run
-                                }
-                                val last = startOfLine + `class`.length
-                                val linkInfo = UnrealClassHyperLinkInfo(model.navigateToClass, UClass(FString(`class`)))
-                                consoleView.hyperlinks.createHyperlink(startOfLine, last, null, linkInfo)
-                            }
-                            run {
-                                if (currentRevision != UnrealPane.revision) {
-                                    return@run
-                                }
-                                val linkInfo = MethodReferenceHyperLinkInfo(model.navigateToMethod, methodReference)
-                                val first = endOfLine - method.length
-                                consoleView.hyperlinks.createHyperlink(first, endOfLine, null, linkInfo)
-                            }
-                        }
-                    }
-                }
-                if (rangeWithIndex.index < allRanges.size - 1 &&
-                        allRanges[rangeWithIndex.index + 1].first != rangeWithIndex.value.last) {
-                    consoleView.print(line.substring(rangeWithIndex.value.last,
-                            allRanges[rangeWithIndex.index + 1].first), NORMAL_OUTPUT)
-                }
+        val allRanges = (unrealLogEvent.bpPathRanges + unrealLogEvent.methodRanges).sortedBy { it.first }
+        val line = unrealLogEvent.text.data
+        if (allRanges.first().first > 0) {
+            consoleView.print(line.substring(0, allRanges[0].first), NORMAL_OUTPUT)
+        }
+
+        for (rangeWithIndex in allRanges.withIndex()) {
+            val match = line.substring(rangeWithIndex.value.first, rangeWithIndex.value.last)
+            if (rangeWithIndex.value in unrealLogEvent.bpPathRanges) {
+                val hyperLinkInfo = BlueprintClassHyperLinkInfo(model.openBlueprint, BlueprintReference(FString(match)))
+                consoleView.printHyperlink(match, hyperLinkInfo)
+            } else {
+                val (`class`, method) = match.split(MethodReference.separator)
+                val methodReference = MethodReference(UClass(FString(`class`)), FString(method))
+
+                val classHyperLinkInfo = UnrealClassHyperLinkInfo(model, methodReference, UClass(FString(`class`)))
```

Comment:
Won't that create a hyperlink which leads nowhere if it's not a valid class::method?

Reply:
Yes, now we will create false hyperlinks sometimes, but it's the only solution I came with to not rely on offset calculation in console text buffer that is very unstable.
I think it's good enough compromise for much simpler code and stable behavior.

Reply:
I see. I think it's an acceptable solution for now.